juicy-html.html

```diff
@@ -145,6 +145,9 @@
                         break;
                 }
             }
+            /* even though it usually doesn't make sense to dispatch an event for a change coming from the outside, 
+            this time it does. Because attributeChangedCallback is asynchronous (due to the polyfill) and we need to know when it was actually called */
+            this.dispatchEvent(new CustomEvent('attribute-changed', {name, value: newVal}));
```

Comment:
That's something I would try to avoid.

It smells like asking for troubles. Could easily result in infinite loops, problems with polymer data-binding, it's throwing more events that are not related to the actual functionality of the element, and it exposes a new public API which could be hard to remove in future.

If we need it only for testing, maybe we could `stub` it with a callback/event?

Reply:
Good point.

Reply:
Apparently, we can't stub it 
![image](https://user-images.githubusercontent.com/17054134/30055078-940fad64-922e-11e7-88ed-ab17d152958c.png)

Reply:
Ping.